spring-cloud-gcp-bigquery/src/main/java/org/springframework/cloud/gcp/bigquery/integration/outbound/BigQueryFileMessageHandler.java

```diff
@@ -0,0 +1,312 @@
+
+		this.formatOptionsExpression =
+				new FunctionExpression<Message>(
+						message -> message.getHeaders().get(BigQuerySpringMessageHeaders.FORMAT_OPTIONS));
+	}
+
+	@Override
+	protected void doInit() {
+		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+	}
+
+	/**
+	 * Sets the SpEL {@link Expression} to evaluate to determine the dataset name.
+	 * @param datasetNameExpression the SpEL expression used to evaluate the dataset name
+	 */
+	public void setDatasetNameExpression(Expression datasetNameExpression) {
+		Assert.notNull(datasetNameExpression, "Dataset name expression can't be null.");
+		this.datasetNameExpression = datasetNameExpression;
+	}
+
+	/**
+	 * Sets the BigQuery dataset name to use.
+	 * @param datasetName name of the BigQuery dataset
+	 */
+	public void setDatasetName(String datasetName) {
+		this.datasetNameExpression = new LiteralExpression(datasetName);
+	}
+
+	/**
+	 * Sets the SpEL {@link Expression} to evaluate to determine the table name.
+	 * @param tableNameExpression the SpEL expression used to evaluate the table name
+	 */
+	public void setTableNameExpression(Expression tableNameExpression) {
+		Assert.notNull(tableNameExpression, "Table name expression must not be null.");
+		this.tableNameExpression = tableNameExpression;
+	}
+
+	/**
+	 * Sets the BigQuery table name to use.
+	 * @param tableName name of the BigQuery table
+	 */
+	public void setTableName(String tableName) {
+		this.tableNameExpression = new LiteralExpression(tableName);
+	}
+
+	/**
+	 * Sets the default {@link FormatOptions} to use for the handler.
+	 * The {@link FormatOptions} describe the type/format of data files being loaded.
+	 * @param formatOptions the format of the data file being loaded
+	 */
+	public void setFormatOptions(FormatOptions formatOptions) {
+		Assert.notNull(formatOptions, "Format options must not be null.");
+		this.formatOptionsExpression = new ValueExpression<>(formatOptions);
+	}
+
+	/**
+	 * Sets the SpEL {@link Expression} used to determine the {@link FormatOptions} for the handler.
+	 * @param formatOptionsExpression the SpEL expression used to evaluate the {@link FormatOptions}
+	 */
+	public void setFormatOptionsExpression(Expression formatOptionsExpression) {
+		Assert.notNull(formatOptionsExpression, "Format options expression cannot be null.");
+		this.formatOptionsExpression = formatOptionsExpression;
+	}
+
+	/**
+	 * Sets the {@link WriteDisposition} which specifies how data should be inserted into
+	 * BigQuery tables.
+	 * @param writeDisposition whether data should be appended or truncated to the BigQuery table
+	 */
+	public void setWriteDisposition(WriteDisposition writeDisposition) {
+		Assert.notNull(writeDisposition, "Write disposition must not be null.");
+		this.writeDisposition = writeDisposition;
+	}
+
+	/**
+	 * Sets whether BigQuery should attempt to autodetect the schema of the data when loading
```

Comment:
mention default behavior?

Reply:
Added defaults in the docs for the more stable options like `sync`, `autodetect`, etc. but I omitted them for the timeouts because I felt that there is a chance the defaults get changed in the future and the doc will get out of sync.